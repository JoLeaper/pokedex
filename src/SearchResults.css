.searchresults {
    width: 100%;
    display: grid;
    grid-template-columns: 25% 25% 25% 25%
}

img {
    height: 100px;
    width: auto;
}

.singularPokemon {
    border: 2px solid black;
    background-color: lightgray;
    margin: 0px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    text-align: center;
    width: auto;
    height: 400px;
}

p {
    margin: 0;
}

li {
    margin: 0;
}