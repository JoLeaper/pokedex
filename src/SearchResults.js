import React, { Component } from 'react'
import './SearchResults.css'

export default class SearchResults extends Component {
    
    render() {
        return (
            <div className='searchresults'>
                 {this.props.pokemonList.map(pokemon => 
                    <li className="singularPokemon" >
                        <h2>{pokemon.pokemon}</h2>
                        <img src={pokemon.url_image} alt={pokemon.pokemon}/>
                    </li >)}
            </div>
        )
    }
}
