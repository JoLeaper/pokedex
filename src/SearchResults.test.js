import React from 'react';
import { shallow } from 'enzyme';
import SearchResults from './SearchResults'
const pokemon = [0];

test('renders search results', () => {
    const wrapper = shallow(<SearchResults pokemonList={pokemon} />);
    expect(wrapper).toMatchSnapshot();
});
