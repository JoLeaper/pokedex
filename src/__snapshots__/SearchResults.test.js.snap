// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders search results 1`] = `
<div
  className="searchresults"
>
  <li
    className="singularPokemon"
  >
    <Link
      to="/pokemon/undefined"
    >
      <h2 />
    </Link>
    <img />
    <p>
      Type: 
        
      / undefined
    </p>
    <p>
      Ability: 
    </p>
    <p>
      Hidden Ability: 
    </p>
    <p>
      Base HP: 
    </p>
    <p>
      Base Attack: 
    </p>
    <p>
      Base Defense: 
    </p>
    <p>
      Base Special Attack: 
    </p>
    <p>
      Base Special Defense: 
    </p>
    <p>
      Base Speed: 
    </p>
  </li>
</div>
`;
